_posts/2017-01-17-Oredev2016.md

```diff
@@ -0,0 +1,696 @@
+
+# A developer's guide to the UX galaxy - Tess Ferrandez
+
+<img class="float-image p-image" src="{{ '/img/oredev/tessferrandez.jpg' | prepend: site.baseurl }}" alt="Tess Ferrandez" style="width: 100px"/>
+
+[Tess Ferrandez](https://www.twitter.com/TessFerrandez) is a full stack developer on the Microsoft Platform.
+She is equally happy debugging nasty backend issues as she is developing apps or working on UI/UX design.
+She has been blogging at [her blogsite](http://blogs.msdn.microsoft.com/Tess) for the better part of her career, and has spoken at lots and lots of conferences around the world.
+
+Software is built into two parts, it is what we see and how we do it.
+In this session Tess talks about what these features should look like.
+If you design 'an order' incorrectly and the user can't figure out to place that order, you have a bad UX design and it could cost your company a lot of money.
+Now, two UX researches at Microsoft deep dived into 100 years of research of signs and symbols and listened to what people define as a good interface.
+At last they categorized every piece into 'Tenets' (attributes of good user interfaces) and 'Traps' (bloopers and things to avoid)
+
+## Tenets : attributes of good use
+We will be using a scanner as example that you use in the supermarket.
+
+
+Understandable
+--------------
+>Ask yourself, what can I do? Is it understandable?
+If you look at the scanner in a supermarket, do you know which button you have to press to save a product?
+It is important to let your users understand what they are interacting with because if they have to read the manual first or ask someone for help, you're doing something wrong.
+
+
+Physically effortless
+----------------------
+>Is it what I am holding quick and comfortable to use?
+When scanning a product you want your users to feel that they can scan the product easily and with no difficulty.
+
+
+Responsive
+----------------------
+> You get immediate feedback, when scanning a product you get an immediate response ( ex: Hey! you scanned something )
+If you don't get an immediate response, the user will keep trying to scan the product and at the end have 10x milk boxes.
+
+Efficient
+----------------------
+> What is the most efficient way in a process.
+The best way to buy 10 milk boxes is to scan the milk 10 times than to write it down 10x with your other hand.
+
+Forgiving
+----------------------
+> I can undo my actions.
+If I scan too many items, I have a minus button to lessen the amount.
+It's a bad practice if there is no button and you have to ask an employee to help you.
+
+Discrete
+----------------------
+> Don't over share, don't overreact.
+A siren on the scanner to point out that you did something wrong.
+
+Protective
+----------------------
+> Always deliver a qualified product with no failure or data loss
+A bad practice is if you fill in a form, go to the next page and realising that you forgot something on the previous page and that your form is blank
```

Comment:
A bad practice for example is that after you just filled in a form, clicked through to the next page and tried to go back to the form to add some missing information only to find out that your form has been completely reset.